src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -94,11 +117,51 @@ def as_strings(self) -> Sequence[str]:
         return keys
 
 
+class UseCaseKeyCollection:
```

Comment:
Same as above, why do we still need `KeyCollection` if we are moving to `UseCaseKeyCollection` ?

Reply:
The rest of the system right now are is still using it, it will get remove once we start using UseCaseKeyCollection